lineage/figures/figure6.py

```diff
@@ -91,9 +96,20 @@ def figureMaker5(ax, distribution_df, wasser_df):
 
     i += 1
     # state accuracy
-    sns.regplot(x="Wasserstein distance", y="Random Index Accuracy", data=wasser_df, label="tHMM", ax=ax[i], lowess=True, marker='+')
-    sns.regplot(x="Wasserstein distance", y="KMeans Accuracy", data=wasser_df, ax=ax[i], label="K-means", lowess=True, marker='+')
+    sns.regplot(x="Wasserstein distance", y="Adjusted Rand Index Accuracy", data=wasser_df, label="tHMM", ax=ax[i], lowess=True, marker='+')
+    sns.regplot(x="Wasserstein distance", y="Baseline Accuracy", data=wasser_df, ax=ax[i], label="Best threshold", lowess=True, marker='+')
     ax[i].set_title("State Assignment Accuracy")
-    ax[i].set_ylabel("Random Index Accuracy [%]")
+    ax[i].set_ylabel("Adjusted Rand Index Accuracy [%]")
     ax[i].set_ylim(bottom=10.0, top=101)
     ax[i].legend()
+
+def classification_threshold(shape, scale1, scale2):
+    """ Given the parameters of the gamma distribution, it provides an analytical threshold for classification.
+    This function is specific to this figure, as the shape parameter is shared and only the scale varies.
+    """
+    if scale1 == scale2:
+        return shape * scale1
+    else:
+        numer = shape * np.log(scale2/scale1)
```

Comment:
What is your reasoning behind this threshold?

My thinking was that we might even be able to get rid of using the data altogether, and just draw the accuracy line from an analytical expression. Maybe that's too ambitious...

Reply:
Oh! I see above that you're finding the point at which they are equal probability. Clever. I'm fine with that.

Reply:
when scale1 == scale2, the denominator becomes zero, and the two distributions are exactly the same, so I assumed the mean of the distribution would be the threshold which I think should result in 50% accuracy.
If you have any other suggestions I can edit this.

I think I can do that by working with the areas under the pdfs. I will work on it.

Reply:
But the scale isn't the same—the shape is the same.

Reply:
Oh I see. That's your fallback. That's fine.